### PR TITLE
Add headless mode using aria-vue lib with (mocha, chai and puppeteer)

### PR DESCRIPTION
**Pull request by aelbore**
- able to run headless mode
- able to run tdd or browse tests in html
- able to run tdd or browse tests using vite
- use aria-vue library or cli run headless and tdd mode
- can use the aria-vue api to run test programatically

Please see https://github.com/aelbore/aria-vue 
for options of the cli

### CODE REVIEW

**Comment by JessicaSachs** [comment]
Sweet!! I will look at this soon.

File: package.json [hunk 1]
```diff
@@ -4,17 +4,20 @@
   "scripts": {
     "dev": "vite",
     "build": "vite build",
-    "tdd": "vite --config=./test/vite.config.test.js"
+    "tdd.vite": "vite --config ./test/vite.config.test.js",
```
Comment (JessicaSachs) [inline package.json:7]:
```suggestion
    "tdd:vite": "vite --config ./test/vite.config.test.js",
```

File: src/App.vue [hunk 2]
```diff
@@ -1,10 +1,12 @@
 <template>
-  <h1>Hello Vite + Vue 3!</h1>
-  <p>Edit ./App.vue to test hot module replacement (HMR).</p>
-  <p>
-    <span>Count is: <span data-testid="button-counter">{{ count }}</span></span>
-    <button @click="count++">increment</button>
-  </p>
+  <div>
```
Comment (JessicaSachs) [inline src/App.vue:2]: This is not required in Vue 3, why add it?

File: test/index.html [hunk 3]
```diff
@@ -1,56 +0,0 @@
-<html>
```
Comment (JessicaSachs) [inline test/index.html:1]: The example at `test/index.html` is broken... with `tdd:vite` you're unable to view your completed test runs with the mocha HTML reporter.

File: test/vite.config.test.js [hunk 4]
```diff
@@ -1,23 +1,10 @@
...
-  app.use(router.routes())
-}
+const { testPlugin, watchPlugin } = require('aria-vue')
 
 module.exports = {
-  // plugins for test
-  configureServer: [ getTestFiles ]
-}
+  configureServer: [
+    testPlugin({
```
Comment (JessicaSachs) [inline test/vite.config.test.js:5]: Can you explain the `/init` route and other koa plugin inside of `aria-vue` that is trying to replace this feature?

File: src/App.vue (hunk 2 above)
Comment (JessicaSachs) [reply src/App.vue:2]: Ah. I see Vue 2 is now in scope. Ok. https://github.com/vitejs/vite/issues/305#issuecomment-636130174

File: test/vite.config.test.js (hunk 4 above)
Comment (aelbore) [reply test/vite.config.test.js:5]: theres some scripts that you need to execute or call before the spec or test files load, like custom plugins of vue-utils, init route will load your scripts before spec files

using CLI
```
aria-vue -H --script ./test/your-custom-plugins.js
```
`--script` option is script that you want to execute before loading the spec files

Using Config
```javascript
    testPlugin({   
      script: './test/plugins.js'
    }),
```

File: test/index.html (hunk 3 above)
Comment (aelbore) [reply test/index.html:1]: By default theres no need for index.html fo your test its serve already in ./node_modules/aria-vue/index.html but you can still customize and create your own index.html.

Using CLI
```
aria-vue --path test-ui --script ./test/your-plugins.js --html ./test/index.html
```
`--path` option is the virtual url path by default its tests 
so you can browse http://localhost:3000/test-ui
`--html` option is where your custom index.html of your test reporter

Using Config
```javascript    
 testPlugin({   
   script: './test/plugins.js',
   path: 'test-ui',
   html: './test/index.html'
  }),
```

Note: on your custom index.html you should load the `setup.js`
```html
 <script type="module" src="/node_modules/aria-vue/setup.js">
```

**Comment by aelbore** [comment]
> Thank you for the PR! My apologies for the late review.
> 
> This PR causes a regression in the following features. I'd love to use it once these are resolved:
> 
> * ability to run my app's index.html (Should only be App.vue) separately from my tests (tests/index.html)
> * configure my test runner and view my tests at `tests/index.html` (Mocha HTML reporter)
> * run all of my tests sequentially
> * run a single test `test/index.html?spec=App.spec.js`

* ability to run my app's index.html (Should only be App.vue) separately from my tests (tests/index.html) 
  Using the configuration you can run your app, so if you `npm run test:vite`
  you can browse your app in http://localhost:3000 if you want view your tests you can browser to
 http://localhost:3000/tests

 * run a single test `test/index.html?spec=App.spec.js`
  yes you can single  http://localhost:3000/tests?spec=App.spec.js

* configure my test runner and view my tests at `tests/index.html` (Mocha HTML reporter)
 yes you can configure your test runner Please see https://github.com/aelbore/aria-vue 
for more options

* run all of my tests sequentially (is this means that you can configure the test in order?)

**Comment by aelbore** [comment]
@JessicaSachs by the way let me know if anything i can improve on the readme file
https://github.com/aelbore/aria-vue

**Comment by JessicaSachs** [comment]
Running all the tests sequentially would mean that when you go to `localhost:3000/tests`, it executes _both_ `HelloWorld.spec.js` and `App.spec.js` and anything that matches a pattern like `*.spec.js` (or .tsx, .jsx)`

That is what the code at https://github.com/JessicaSachs/vite-component-test-starter/pull/5/files#diff-ab32b98942d40992faa2e5e3f8ac9563L41-L42 does when it loads all of the spec files and then waits to run mocha.

**Comment by JessicaSachs** [comment]
OK! Super. I have sequential test execution working. I have a few more things I've found... one of which should be fixed in `aria-vue`

This was fixed in the `test/index.html` file by deleting the Vue Devtools toast. Can you port this fix to `aria-vue`?

Screenshot 2020-06-02 21 16 37

**Comment by aelbore** [comment]
> OK! Super. I have sequential test execution working. I have a few more things I've found... one of which should be fixed in `aria-vue`
> 
> 
> 
> This was fixed in the `test/index.html` file by deleting the Vue Devtools toast. Can you port this fix to `aria-vue`?
> 
> 
> 
> Screenshot 2020-06-02 21 16 37
> 
> 

Oh i thought i deleted it in setup.js maybe a typo i will check it het it back. Thanks

**Comment by JessicaSachs** [comment]
NP! Once that's in, we can merge. I just fixed some docs and missing newlines

**Comment by aelbore** [comment]
> NP! Once that's in, we can merge. I just fixed some docs and missing newlines

Just fixed the with you version of aria-vue@0.3.10

**Pull request merged**